docs/resources/samples/combine-worksheets-into-single-workbook.md

```diff
@@ -0,0 +1,103 @@
+/**
+ * This script creates a new worksheet in the current workbook for each WorksheetData object provided.
+ */
+function main(workbook: ExcelScript.Workbook, workbookName: string, worksheetInformation: WorksheetData[])
+{
+  // Add each new worksheet.
+  worksheetInformation.forEach((value) => {
+    let sheet = workbook.addWorksheet(`${workbookName}.${value.name}`);
+
+    // If there was any data in the worksheet, add it to a new range.
+    if (value.data) {
+      let range = sheet.getRangeByIndexes(0, 0, value.data.length, value.data[0].length);
+      range.setValues(value.data);
+    }
+  });
+}
+
+// An interface to pass the worksheet name and cell values through a flow.
+interface WorksheetData {
+  name: string;
+  data: string[][];
+}
+```
+
+## Power Automate flow: Combine worksheets into a single workbook
+
+1. Sign into [Power Automate](https://flow.microsoft.com) and create a new **Instant cloud flow**.
+1. Choose **Manually trigger a flow** and select **Create**.
+1. Get all the files in the folder. In this example, we'll use a folder named "output". Add a **New step** that uses the **OneDrive for Business** connector and the **List files in folder** action. Provide the folder path that contains the .csv files.
+    * **Folder**: /output
+
+    :::image type="content" source="../../images/combine-worksheets-flow-1.png" alt-text="The completed OneDrive for Business connector in Power Automate.":::
+1. Run the **Return Worksheet Data** script to get all the data from each of the workbooks. Add the **Excel Online (Business)** connector with the **Run script** action. Use the following values for the action. Note that when you add the *Id* for the file, Power Automate will wrap the action in an **Apply to each** control, so the action will be performed on every file.
+    * **Location**: OneDrive for Business
+    * **Document Library**: OneDrive
+    * **File**: *Id* (dynamic content from **List files in folder**)
+    * **Script**: Return Worksheet Data
+1. Run the **Add Worksheets** script on the new Excel file you created. This will add the data from all the other workbooks. After the previous **Run script** action and inside the **Apply to each** control, add an **Excel Online (Business)** connector with the **Run script** action. Use the following values for the action.
+    * **Location**: OneDrive for Business
+    * **Document Library**: OneDrive
+    * **File**: Your file
+    * **Script**: Add Worksheets
+    * **workbookName**: *Name* (dynamic content from **List files in folder**)
+    * **worksheetInformation**: *result* (dynamic content from **Run script**)
+
+    :::image type="content" source="../../images/combine-worksheets-flow-2.png" alt-text="The two Run script actions inside the Apply to each control.":::
+1. Save the flow. Use the **Test** button on the flow editor page or run the flow through your **My flows** tab. Be sure to allow access when prompted.
+1. Your Excel file should now have new worksheets.
```

Comment:
My flow failed :(

Reply:
My flow failed again, this time with a BadGateway error: 
![image](https://user-images.githubusercontent.com/7977793/132779339-ec855a9f-3729-4016-9672-0e4536033ae8.png)

Reply:
Hmm, I'm guessing a Bad Gateway is unrelated to the script itself:
https://powerusers.microsoft.com/t5/General-Power-Automate/502-BadGateway-Error-When-Power-Automate-Changes-Document-To/td-p/741296